src/drivers/input/button/stm32cube/stm32cube_button.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * @file
+ *
+ * @date 15.04.2020
+ * @author Alexander Kalmuk
+ */
+
+#include <util/log.h>
+
+#include <errno.h>
+#include <assert.h>
+
+#include <kernel/irq.h>
+#include <drivers/input/input_dev.h>
+
+#include <drivers/gpio/gpio.h>
+
+#include <embox/unit.h>
+#include <framework/mod/options.h>
+
+
+EMBOX_UNIT_INIT(stm32cube_bt_init);
+
+#define USER_BUTTON_PORT       OPTION_GET(NUMBER,pin_port)
+#define USER_BUTTON_PIN_MASK   (1 << OPTION_GET(NUMBER,pin_num))
+
+struct stm32cube_bt_indev {
+	struct input_dev input_dev;
+};
+
+static struct stm32cube_bt_indev stm32cube_bt_dev;
+
+static void stm32cbe_bt_poll(struct input_dev *dev) {
+	struct input_event ev;
+
+	ev.type = USER_BUTTON_PRESSED;
+	ev.value = 0;
+
+	input_dev_report_event(dev, &ev);
```

Comment:
It would be nice if we could monitor both PRESS and RELEASE events on a button.

Reply:
Thanks. Done